rln-cli/src/examples/relay.rs

```diff
@@ -0,0 +1,303 @@
+                    self.handle_duplicate_message_id(previous_proof.clone(), proof_data)?;
+                    return Ok(());
+                }
+                self.used_nullifiers.insert(nullifier_key, proof_data);
+                println!("Message verified and accepted");
+            }
+            Ok(false) => {
+                println!("Verification failed: message_id must be unique within the epoch and satisfy 0 <= message_id < MESSAGE_LIMIT: {MESSAGE_LIMIT}");
+            }
+            Err(err) => return Err(err),
+        }
+        Ok(())
+    }
+
+    fn handle_duplicate_message_id(
+        &mut self,
+        previous_proof: Vec<u8>,
+        current_proof: Vec<u8>,
+    ) -> Result<()> {
+        let x = &current_proof[192..224];
+        let y = &current_proof[224..256];
+
+        let prev_x = &previous_proof[192..224];
+        let prev_y = &previous_proof[224..256];
+        if x == prev_x && y == prev_y {
+            return Err(eyre!("this exact message and signal has already been sent"));
+        }
+
+        let mut proof1 = Cursor::new(previous_proof);
+        let mut proof2 = Cursor::new(current_proof);
+        let mut output = Cursor::new(Vec::new());
+
+        match self
+            .rln
+            .recover_id_secret(&mut proof1, &mut proof2, &mut output)
+        {
+            Ok(_) => {
+                let output_data = output.into_inner();
+                let (leaked_identity_secret_hash, _) = bytes_le_to_fr(&output_data);
+
+                if let Some((user_index, identity)) = self
+                    .local_identities
+                    .iter()
+                    .find(|(_, identity)| {
+                        identity.identity_secret_hash == leaked_identity_secret_hash
+                    })
+                    .map(|(index, identity)| (*index, identity))
+                {
+                    let real_identity_secret_hash = identity.identity_secret_hash;
+                    if leaked_identity_secret_hash != real_identity_secret_hash {
+                        Err(eyre!("identity secret hash mismatch {leaked_identity_secret_hash} != {real_identity_secret_hash}"))
+                    } else {
+                        println!("DUPLICATE message ID detected! Reveal identity secret hash: {leaked_identity_secret_hash}");
+                        self.local_identities.remove(&user_index);
+                        self.rln.delete_leaf(user_index)?;
+                        println!("User index {user_index} has been SLASHED");
+                        Ok(())
+                    }
+                } else {
+                    Err(eyre!(
+                        "user identity secret hash {leaked_identity_secret_hash} not found"
+                    ))
+                }
+            }
+            Err(err) => Err(eyre!("Failed to recover identity secret: {err}")),
+        }
+    }
+}
+
+fn main() -> Result<()> {
+    let mut rln_system = RLNSystem::new()?;
```

Comment:
When I run this command `cargo run --example relay` It takes a lot of time to load zkey (that's expected), but in cli it looks like it stuck. So I suggest to add here comment that you load data and create instance 

And same for stateless

Reply:
could you also make the same in stateless file?